OrbitLinuxTracing/TracerThread.cpp

```diff
@@ -181,7 +181,19 @@ bool TracerThread::OpenSampling(const std::vector<int32_t>& cpus) {
   std::vector<int> sampling_tracing_fds;
   std::vector<PerfEventRingBuffer> sampling_ring_buffers;
   for (int32_t cpu : cpus) {
-    int sampling_fd = sample_event_open(sampling_period_ns_, -1, cpu);
+    int sampling_fd;
+    switch (tracing_options_.sampling_method) {
+      case kFramePointers:
+        sampling_fd = callchain_sample_event_open(sampling_period_ns_, -1, cpu);
+        break;
+      case kDwarf:
+        sampling_fd = sample_event_open(sampling_period_ns_, -1, cpu);
+        break;
+      case kOff:
+        ERROR("Sampling is off");
```

Comment:
Why is this an error?

Reply:
Because we should never ever reach that program point. If kOff is set, then we will not enable sampling and would not enter this method

Reply:
This is basically a failed assertion, you should use `FATAL`.

Reply:
I changed it to FATAL. Thanks for the suggestion.